jena-fuseki2/jena-fuseki-embedded/src/main/java/org/apache/jena/fuseki/embedded/FusekiServer.java

```diff
@@ -455,30 +527,36 @@ private static void setMimeTypes(ServletContextHandler context) {
 
         private void servlets(ServletContextHandler context) {
             // Fuseki dataset services filter
+            // This goes as the filter at the end of any filter chaining.
             FusekiFilter ff = new FusekiFilter();
-            FilterHolder h = new FilterHolder(ff);
-            context.addFilter(h, "/*", null);
-
-            other.forEach(p->addServlet(context, p.getLeft(), p.getRight()));
+            addFilter(context, "/*", ff);
             
             if ( withStats )
                 addServlet(context, "/$/stats", new ActionStats());
             if ( withPing )
                 addServlet(context, "/$/ping", new ActionPing());
             
+            servlets.forEach(p->addServlet(context, p.getLeft(), p.getRight()));
+            filters.forEach (p-> addFilter(context, p.getLeft(), p.getRight()));
+
             if ( staticContentDir != null ) {
                 DefaultServlet staticServlet = new DefaultServlet();
                 ServletHolder staticContent = new ServletHolder(staticServlet);
                 staticContent.setInitParameter("resourceBase", staticContentDir);
                 context.addServlet(staticContent, "/");
             }
         }
-
+        
```

Comment:
Nit-picking again! Unnecessary spaces...

Reply:
>simply turning on the save action to remove trailing spaces is likely to cause such formatting changes on files with functional changes.

Agreed, I used the save action in Eclipse for a while and regretted later.

>Previously, keeping formatting changes separate from functional changes has been requested. (I'm neutral.)

I'm ok with either too. But my point here was that this case wasn't like the save action... at least from looking at the github diff, it appears there was no spaces before, and these were added in this commit only. But as I said, nit-pick. And nit-picks like this one can cause distraction, so feel free to ignore or hide :-)

Reply:
These leading space arise in Eclipse. After a closing "}" the cursor is padded to the same depth as the "}". Is there a way to change that without using a save action?